compiler/luci-interpreter/src/kernels/SelectV2.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ * Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kernels/SelectV2.h"
+#include "kernels/Utils.h"
+
+#include <tensorflow/lite/kernels/internal/reference/reference_ops.h>
+// TODO use select.h when version up
+// #include <tensorflow/lite/kernels/internal/reference/select.h>
+
+#include <stdexcept>
+
+namespace luci_interpreter
+{
+namespace kernels
+{
+
+SelectV2::SelectV2(const Tensor *condition, const Tensor *t, const Tensor *e, Tensor *output)
+  : Kernel({condition, t, e}, {output})
+{
+  _has_low_rank_input_condition = false;
+}
+
+void SelectV2::configure()
+{
+  LUCI_INTERPRETER_CHECK(condition()->element_type() == DataType::BOOL);
+  LUCI_INTERPRETER_CHECK(t()->element_type() == e()->element_type());
+  LUCI_INTERPRETER_CHECK(t()->element_type() == output()->element_type());
+
+  auto cond_shape = condition()->shape();
+  auto cond_num_dims = cond_shape.num_dims();
+  auto t_shape = t()->shape();
+  auto e_shape = e()->shape();
+
+  bool is_input_condition_scalar = cond_num_dims == 0;
+  bool has_rank_one_input_condition = cond_num_dims == 1 && cond_shape.dim(0) == t_shape.dim(0);
+
+  _has_low_rank_input_condition =
+    (is_input_condition_scalar || has_rank_one_input_condition) && t_shape == e_shape;
```

Comment:
This behavior is different from tflite kernel.

```suggestion
  _has_low_rank_input_condition =
    is_input_condition_scalar || has_rank_one_input_condition;
```

`t_shape == e_shape` is necessary? It seems that `has_low_rank_input_condition` is only used by select v1.

Reply:
> `t_shape == e_shape` is necessary?

Yes, it is necessary if select v2 kernel uses `RankOneSelect()`.

> `has_low_rank_input_condition` is only used by select v1.

I just implemented it following below conditions. 

### TFL SelectV2 op

https://www.tensorflow.org/mlir/tfl_ops#tflselect_v2_tflselectv2op

Select values of 'x' if the corresponding value of 'condition' is true or the value of 'y' if false. There are valid condition input sizes:

1. Either the same shape (in which case the select is elementwise), or
2. Broadcastable shapes between 'condition', 'x' and 'y'.

Anyway, I will change this kernel to use only `BroadcastSelect4DSlow()` as your advice.